src/MusicPlaylistKernel.java

```diff
@@ -0,0 +1,39 @@
+import components.standard.Standard;
+
+/**
+ * Music Playlist component with primary methods.
+ *
+ */
+public interface MusicPlaylistKernel extends Standard<MusicPlaylist> {
+    /**
+     * Adds {@code (song, artist)} to {@code this}.
+     *
+     * @param song
+     *            the {@code song} to be added
+     * @param artist
+     *            the {@code artist} to be added
+     * @updates this
+     * @ensures |this| = |#this| + 1
+     */
+    void add(String song, String artist);
+
+    /**
+     * Removes {@code song} from {@code this} and reports the song.
+     *
+     * @param song
+     *            the {@code song} to be removed
+     * @updates this
+     * @ensures |this| = |#this| - 1
+     * @requires |this| > 0 and song is in DOMAIN (this)
```

Comment:
Note that if the precondition is that the song exists, then you need a method that checks (e.g., contains).